CSharp/Microsoft.Bot.Builder.Teams/Middlewares/TeamsTenantFilteringMiddleware.cs

```diff
@@ -51,23 +51,17 @@ public TeamsTenantFilteringMiddleware(IEnumerable<string> allowedTenantIds)
         /// </remarks>
         /// <seealso cref="ITurnContext" />
         /// <seealso cref="Schema.IActivity" />
-        public Task OnTurnAsync(ITurnContext turnContext, NextDelegate next, CancellationToken cancellationToken = default(CancellationToken))
+        public async Task OnTurnAsync(ITurnContext turnContext, NextDelegate next, CancellationToken cancellationToken = default(CancellationToken))
         {
-            // Ignoring cases where ChannelData is missing or does not contain TenantId.
-            if (turnContext.Activity.ChannelData != null)
-            {
-                TeamsChannelData teamsChannelData = turnContext.Activity.GetChannelData<TeamsChannelData>();
+            TeamsChannelData teamsChannelData = turnContext.Activity.GetChannelData<TeamsChannelData>();
+            string tenantId = teamsChannelData?.Tenant?.Id;
 
-                if (!string.IsNullOrEmpty(teamsChannelData?.Tenant?.Id))
-                {
-                    if (!this.tenantMap.ContainsKey(teamsChannelData.Tenant.Id))
-                    {
-                        throw new UnauthorizedAccessException("Tenant Id '" + teamsChannelData.Tenant.Id + "' is not allowed access.");
-                    }
-                }
+            if (string.IsNullOrEmpty(tenantId) || !this.tenantMap.ContainsKey(tenantId))
+            {
+                throw new UnauthorizedAccessException("Tenant Id '" + tenantId + "' is not allowed access.");
```

Comment:
This will throw a weird error in case TenantId is empty. Tenant Id '' is not allowed access. Instead let's throw an error saying tenant Id was missing.